pandas/core/nanops.py

```diff
@@ -1784,11 +1790,9 @@ def na_accum_func(values: ArrayLike, accum_func, *, skipna: bool) -> ArrayLike:
             # DatetimeArray/TimedeltaArray
             # TODO: have this case go through a DTA method?
             # For DatetimeTZDtype, view result as M8[ns]
+            values = cast("DatetimeArray | TimedeltaArray", values)
```

Comment:
i think just DatetimeArray (comment on L1790 should reflect this too); OK for follow-up.

Reply:
actually it's probably best to remove the cast and keep the "fix later" ignore. A 3rd party EA subclass could satisfy `values.dtype.kind in ["m", "M"]` and would not have a `_simple_new` method as that's not a requirement of the EA api.